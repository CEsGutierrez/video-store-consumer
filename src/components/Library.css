.libraryList {
  display: flex;
  flex-wrap: wrap;
}