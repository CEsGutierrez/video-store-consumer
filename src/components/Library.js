import React from 'react';
import Movie from './Movie';
// import PropTypes from 'prop-types';

const parseMovies = (movies, selectMovieCallback) => {
  return movies.map((movie) => {
    return <Movie
      key={movie.id}
      {...movie}
      selectMovieCallback={selectMovieCallback}
      />
  });
}

const Library = ({movieList, selectMovieCallback, resetMessageCallback}) => {
  return(
    <div>
      {resetMessageCallback()}
      {parseMovies(movieList, selectMovieCallback)}
    </div>
  )
}

export default Library;