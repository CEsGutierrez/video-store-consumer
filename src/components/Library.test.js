import React from 'react';
import { render, cleanup } from '@testing-library/react'
import Library from './Library';

describe('Library', () => {
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <Library
      movieList={[]}
      selectMovieCallback={() => { }}
      resetMessageCallback={() => { }}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });

});