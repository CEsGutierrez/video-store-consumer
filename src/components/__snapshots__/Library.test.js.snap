// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
